Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -0,0 +1,1185 @@
+    throw std::runtime_error("CompareWorkspaces::processGroups - Neither "
+                             "input is a WorkspaceGroup. This is a logical "
+                             "error in the code.");
+  } else if (!ws1 || !ws2) {
+    recordMismatch(
+        "Type mismatch. One workspace is a group, the other is not.");
+  }
+
+  setProperty("Result", m_Result);
+  setProperty("Messages", m_Messages);
+
+  // Store output workspace in AnalysisDataService
+  if (!isChild())
+    this->store();
+
+  setExecuted(true);
+  notificationCenter().postNotification(
+      new FinishedNotification(this, this->isExecuted()));
+
+  return true;
+}
+
+//----------------------------------------------------------------------------------------------
+/**
+ * @brief CompareWorkspaces::processGroups
+ * @param groupOne
+ * @param groupTwo
+ */
+void CompareWorkspaces::processGroups(
+    boost::shared_ptr<const API::WorkspaceGroup> groupOne,
+    boost::shared_ptr<const API::WorkspaceGroup> groupTwo) {
+
+  // Check their sizes
+  const size_t totalNum = static_cast<size_t>(groupOne->getNumberOfEntries());
+  if (groupOne->getNumberOfEntries() != groupTwo->getNumberOfEntries()) {
+    recordMismatch("GroupWorkspaces size mismatch.");
+    return;
+  }
+
+  // See if there are any other properties that require setting
+  const std::vector<Property *> &allProps = this->getProperties();
+  std::vector<Property *> nonDefaultProps;
+  nonDefaultProps.reserve(allProps.size());
+  for (size_t i = 0; i < allProps.size(); ++i) {
+    Property *p = allProps[i];
+    const std::string &propName = p->name();
+    // Skip those not set and the input workspaces
+    if (p->isDefault() || propName == "Workspace1" || propName == "Workspace2")
+      continue;
+    nonDefaultProps.push_back(p);
+  }
+  const size_t numNonDefault = nonDefaultProps.size();
+
+  const double progressFraction = 1.0 / static_cast<double>(totalNum);
+  std::vector<std::string> namesOne = groupOne->getNames();
+  std::vector<std::string> namesTwo = groupTwo->getNames();
+  for (size_t i = 0; i < totalNum; ++i) {
+    // We should use an algorithm for each so that the output properties are
+    // reset properly
+    Algorithm_sptr checker = this->createChildAlgorithm(
+        this->name(), progressFraction * (double)i,
+        progressFraction * (double)(i + 1), false, this->version());
+    checker->setPropertyValue("Workspace1", namesOne[i]);
+    checker->setPropertyValue("Workspace2", namesTwo[i]);
+    for (size_t j = 0; j < numNonDefault; ++j) {
+      Property *p = nonDefaultProps[j];
+      checker->setPropertyValue(p->name(), p->value());
+    }
+    checker->execute();
+
+    bool success = checker->getProperty("Result");
+    if (!success) {
+      m_Result = false;
+
+      ITableWorkspace_sptr table = checker->getProperty("Messages");
+      recordMismatch(table->cell<std::string>(0, 0) + ". Inputs=[" +
+                     namesOne[i] + "," + namesTwo[i] + "]");
```

Comment:
Wouldn't it be better to put the inputs into separate columns? In that case the user wouldn't need to parse the message to extract the workspace names.